sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -435,7 +440,7 @@ def _initial_request(self):
 
                 if self._end_range is not None:
                     # Use the end range index unless it is over the end of the file
-                    self.size = min(self._file_size, self._end_range - self._start_range + 1)
+                    self.size = min(self._file_size - self._start_range, self._end_range - self._start_range + 1)
```

Comment:
`self._start_range` cannot be `None` if `self._end_range` is not `None` because on `download_blob` if you specify and end, we enforce that you must provide a start. (Its actually an `offset` and `length` and you can't have `length` without an `offset`).

That being said, MyPy does not know this and so in the typing PR that Vincent is working on, I believe we add an explicit check for `self._start_range` as well but I'll leave that out for now.